### DISCUSSION TITLE
Unable to build moose.

### DISCUSSION THREAD

**Discussion by Edward-Eth** [idaholab/moose#22369] [Q&A General] [answered]

**Comment by GiudGiud** [comment, marked as answer]
Hello

This is indeed the problem, you need to update libmesh. 
Maybe there is a LIBMESH_DIR variable set that is preventing finidng the new version?

Can you remove moose/libmesh then try to update again?
what does echo $LIBMESH_DIR return?

Guillaume

**Reply by Edward-Eth** [reply]
.

**Reply by GiudGiud** [reply]
no this is very problematic.
You can either have libmesh ithrough conda or through a manual install with the update_and_rebuild script not both

If you want to use conda, just mamba update --all

**Reply by Edward-Eth** [reply]
Is there anyway to remove the libmesh installed by update_and_rebuild?

**Reply by GiudGiud** [reply]
rm moose/libmesh -rf

**Reply by Edward-Eth** [reply]
I have performed this and mamba update --all and still get the same errors when I make -j8, even after a clobberall

**Reply by GiudGiud** [reply]
ok then your mamba is not updating properly
this can happen as dependency resolution can prevent updates.

Unless mamba list shows you have a libmesh from September, I would remove the conda environment and re-install it

**Reply by Edward-Eth** [reply]
Yeah my mamba wasn't updating the libmesh version for some reason, removing the environment and recreating it has fixed the issue. Thanks for the help!
